tests/test_tensorflow_data_api.py

```diff
@@ -28,153 +21,188 @@
 ROWS = 1000
 
 
-@pytest.fixture(
-    params=[
-        {
-            "input_shape": (10,),
-        },
-        {
-            "input_shape": (10, 3),
-        },
-        {
-            "input_shape": (10, 10, 3),
-        },
-    ]
+@pytest.mark.parametrize(
+    "input_shape",
+    [
+        (10,),
+        (10, 3),
+        (10, 10, 3),
+    ],
+)
+# We test for single and multiple attributes
+@pytest.mark.parametrize(
+    "num_of_attributes",
+    [1, 2, 3],
 )
-def model(request):
```

Comment:
I assume that we don't need to test the model fit here with different input shapes?

Reply:
We don't need to check any model fit here I think. We check generator's output in terms of shapes and types, so I guess we are good ;)